docs/06-contributors/999-rfcs/2819-stdlib.md

```diff
@@ -0,0 +1,115 @@
+# #2819 - Wing Utility Library
+
+- **Author(s):**: @eladb
+- **Submission Date**: 2023-06-07
+- **Stage**: Proposal
+- **Stage Date**: 2023-06-07
+
+> A specification for the Wing SDK utility libraries.
+
+## Design
+
+The Wing SDK includes a set of submodules which offer a set of useful functionality for building
+cloud applications.
+
+> This is just an initial spec and will likely change in the future.
+
+For each module, we show a list of function declarations. For each function, we specify which phase this function
+is bound to: `inflight` (inflight only - async), `preflight` (preflight only - synchronous) or
+`preflight inflight` (for phase-independent).
+
+## `util` module
+
+```js
+// read environment variables
+preflight inflight env(name: str): str;
+preflight inflight tryEnv(name: str): str?;
+
+// sleep
+inflight sleep(d: duration): void; // inflight only!
+
+// hash/random/uid generation
+preflight inflight uuidv4(): str;
+preflight inflight sha256(data: str): str;
+preflight inflight nanoid(): str; // https://www.npmjs.com/package/nanoid
```

Comment:
I'm not sure if there's a large benefit to bundling these in, given there are so many of them out there and when to use each is less clear cut than e.g. math functions. I feel like by including some hashing / ID generation libraries, we are conveying to users that "hashing / uid generation is a domain of responsibility of Wing's standard library" in which case we might as well include all of them -- md5, uuidv1, uuidv5, sha1, sha512, bcrypt, and so on.

Reply:
I think all of these (and more) are commonly used in cloud programming and I wanted to sketch this direction. Of course this is very partial.